x/ccv/provider/keeper/gov_hook.go

```diff
@@ -0,0 +1,101 @@
+package keeper
+
+import (
+	"fmt"
+
+	"github.com/cosmos/gogoproto/proto"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	govkeeper "github.com/cosmos/cosmos-sdk/x/gov/keeper"
+	sdkgov "github.com/cosmos/cosmos-sdk/x/gov/types"
+	v1 "github.com/cosmos/cosmos-sdk/x/gov/types/v1"
+
+	"github.com/cosmos/interchain-security/v3/x/ccv/provider/types"
+)
+
+type GovHooks struct {
+	gk *govkeeper.Keeper
+	k  *Keeper
+}
+
+// Implements GovHooks interface
+// GovHooks exist in cosmos-sdk/x/gov/keeper/hooks.go of v0.45.16-lsm-ics and on
+var _ sdkgov.GovHooks = GovHooks{}
+
+func (k *Keeper) GovHooks(gk *govkeeper.Keeper) GovHooks {
+	return GovHooks{
+		gk: gk,
+		k:  k,
+	}
+}
+
+// AfterProposalSubmission - call hook if registered
+// After consumerAddition proposal submission, the consumer chainID is stored
+func (gh GovHooks) AfterProposalSubmission(ctx sdk.Context, proposalID uint64) {
+	p, ok := gh.gk.GetProposal(ctx, proposalID)
+	if !ok {
+		panic(fmt.Errorf("failed to get proposal %d in  gov hook", proposalID))
+	}
+	msgs := p.GetMessages()
+
+	for _, msg := range msgs {
+		var msgLegacyContent v1.MsgExecLegacyContent
+		err := proto.Unmarshal(msg.Value, &msgLegacyContent)
+		if err != nil {
+			panic(fmt.Errorf("failed to unmarshal proposal content in gov hook: %w", err))
+		}
+
+		// if the proposal is not ConsumerAdditionProposal, continue
+		if msgLegacyContent.Content.TypeUrl != "/interchain_security.ccv.provider.v1.ConsumerAdditionProposal" {
```

Comment:
Can we revert this to what it was? I did not realize we would need to hardcode a `TypeUrl` string.

Unmarshalling at line 54 should be enough.